server-spi-private/src/main/java/org/keycloak/email/EmailTemplateProvider.java

```diff
@@ -47,6 +49,16 @@
     public void sendPasswordReset(String link, long expirationInMinutes) throws EmailException;
 
     /**
+     * Test SMTP connection with current logged in user
+     *
+     * @param config
+     * @param user
+     * @param realm
+     * @throws EmailException
+     */
+    public void sendSmtpTestEmail(Map<String, String> config, UserModel user, RealmModel realm) throws EmailException;
```

Comment:
You don't need the RealmModel here as that can be retrieved from session.getContext().getRealm()